### PR TITLE
Added recipe for pomegranate

### PR DESCRIPTION
**Pull request by micknudsen**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by micknudsen** [comment]
@conda-forge/help-python I am having troubles figuring out what is wrong here. The error seems to come from something in `conda build` and not in my recipe. Any ideas where to start trouble-shooting?

Thanks!

```
Traceback (most recent call last):
  File "/home/conda/.ci_support/build_all.py", line 131, in <module>
    build_all(args.recipes_dir, args.arch)
  File "/home/conda/.ci_support/build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File "/home/conda/.ci_support/build_all.py", line 121, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2313, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1322, in build
    output_metas = expand_outputs([(m, need_source_download, need_reparse_in_env)])
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 738, in expand_outputs
    for (output_dict, m) in _m.copy().get_output_metadata_set(permit_unsatisfiable_variants=False):
  File "/opt/conda/lib/python3.7/site-packages/conda_build/metadata.py", line 2047, in get_output_metadata_set
    bypass_env_check=bypass_env_check)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/metadata.py", line 721, in finalize_outputs_pass
    permit_unsatisfiable_variants=permit_unsatisfiable_variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 519, in finalize_metadata
    exclude_pattern)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 404, in add_upstream_pins
    permit_unsatisfiable_variants, exclude_pattern)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 373, in _read_upstream_pin_files
    extra_run_specs = get_upstream_pins(m, actions, env)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 361, in get_upstream_pins
    specs = _read_specs_from_package(loc, dist)
  File "/opt/conda/lib/python3.7/site-packages/conda/exports.py", line 198, in __call__
    value = self.func(*args, **kw)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/render.py", line 256, in _read_specs_from_package
    if not specs and os.path.isfile(pkg_loc):
  File "/opt/conda/lib/python3.7/genericpath.py", line 30, in isfile
    st = os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType
```

**Comment by scopatz** [comment]
yes this does come from conda-build

**Comment by micknudsen** [comment]
Same issue as in #8706. In this case, though, it happens for all builds.

@conda-forge/staged is the only solution to wait for an updated version of `conda build`?

**Comment by micknudsen** [comment]
Same issue as in #8706. In this case, though, it happens for all builds.

@conda-forge/staged-recipes is the only solution to wait for an updated version of conda build?

**Comment by micknudsen** [comment]
@scopatz Do you know of any possible way to get around this issue?

**Comment by scopatz** [comment]
At this exact moment, no.  We are still trying to debug this issue.  Though I think a PR to fix it in conda-build would help.

**Comment by scopatz** [comment]
Potential fix at https://github.com/conda/conda-build/pull/3590

**Comment by scopatz** [comment]
Restarted and now this looks like it has legitimate faliures

**Comment by micknudsen** [comment]
@scopatz Thanks for helping me getting further! 👍

File: recipes/pomegranate/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "0ff16e3fbed0047befbefc4a1bad5e612703b3f5f74451725c3421b931dc42c2"
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
+
+requirements:
+  build:
+    - cython
```
Comment (scopatz) [inline recipes/pomegranate/meta.yaml]: You probably need compilers

File: recipes/pomegranate/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  host:
+    - joblib >=0.9.0b4
+    - networkx >=2.0
+    - numpy >=1.8.0
+    - pip
+    - python
+    - pyyaml
+    - scipy >=0.17.0
+  run:
+    - cython
```
Comment (scopatz) [inline recipes/pomegranate/meta.yaml]: Is cython really a runtime requirement?

File: recipes/pomegranate/meta.yaml (hunk 1 above)
Comment (scopatz) [reply recipes/pomegranate/meta.yaml]: cython probably shouldn't be in host, I don't think

File: recipes/pomegranate/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: "0ff16e3fbed0047befbefc4a1bad5e612703b3f5f74451725c3421b931dc42c2"
+
+build:
+  number: 0
+  script: python -m pip install --no-deps --ignore-installed .
```
Comment (scopatz) [inline recipes/pomegranate/meta.yaml]: This should probably be the same as in the example

**Comment by micknudsen** [comment]
@conda-forge/help-python Does anybody know how to address an error like this?

> RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml  url=https://pypi.org/simple/joblib/

I have added `joblib` as build, host, and run requirement, but it still appears to be missing.

File: recipes/pomegranate/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://github.com/jmschrei/{{ name }}/archive/v{{ version }}.tar.gz"
+  sha256: "70566e06da07a95d6be1b6b76acf12d149e0484c48173d5a6493bc85bcc62e0b"
+
+build:
+  number: 0
+  script: "{{ PYTHON }} setup.py install"
```
Comment (beckermr) [inline recipes/pomegranate/meta.yaml]: This build string is wrong. See the example recipe.

**Comment by micknudsen** [comment]
@beckermr @isuruf Thank you very much for your assistance!

**Pull request merged**